crates/lang/ir/src/ir/attrs.rs

```diff
@@ -580,24 +580,42 @@ impl TryFrom<syn::NestedMeta> for AttributeArg {
                                 ).map_err(|_| {
                                     format_err_spanned!(
                                         meta,
-                                        "invalid selector bytes"
+                                        "invalid selector - a selector must consist of four bytes in hex (e.g. `selector = \"0xCAFEBABE\"`)"
                                     )
                                 })?;
                                 let str = lit_str.value();
-                                let cap = regex.captures(&str).unwrap();
+                                let cap = regex.captures(&str)
+                                .ok_or_else(||
+                                   format_err_spanned!(
+                                        meta,
+                                        "invalid selector - a selector must consist of four bytes in hex (e.g. `selector = \"0xCAFEBABE\"`)"
+                                    )
+                                )?;
                                 let selector_bytes = [
-                                    u8::from_str_radix(&cap[1], 16).expect(
-                                        "encountered non-hex digit at position 0",
-                                    ),
-                                    u8::from_str_radix(&cap[2], 16).expect(
-                                        "encountered non-hex digit at position 1",
-                                    ),
-                                    u8::from_str_radix(&cap[3], 16).expect(
-                                        "encountered non-hex digit at position 2",
-                                    ),
-                                    u8::from_str_radix(&cap[4], 16).expect(
-                                        "encountered non-hex digit at position 3",
-                                    ),
+                                    u8::from_str_radix(&cap[1], 16).map_err(|_| {
+                                        format_err_spanned!(
+                                            meta,
+                                            "encountered non-hex digit at position 0",
+                                        )
```

Comment:
Would be nice if we could factor this into its own function with some index parameter because of the code dupe.